components/clientComponents/globals/CaptchaForm/CaptchaForm.tsx

```diff
@@ -0,0 +1,131 @@
+// suspicious users.
+export const CaptchaForm = ({
+  blockableMode = false,
+  hCaptchaSiteKey = "",
+  children,
+  onSubmit,
+  lang,
+  id = "",
+  dataTestId = "",
+  className = "",
+  noValidate = true,
+}: {
+  hCaptchaSiteKey: string | undefined;
+  // Determines whether or not to block a user marked as a bot by siteverify from submitting a form
+  blockableMode?: boolean;
+  children: React.ReactNode;
+  onSubmit: (e?: FormEvent<HTMLFormElement>) => void;
+  lang: string;
+  id?: string;
+  dataTestId?: string;
+  className?: string;
+  noValidate?: boolean;
+}) => {
+  const router = useRouter();
+
+  const { getFlag } = useFeatureFlags();
+  const captchaEnabled = hCaptchaEnabled(getFlag("hCaptcha"), hCaptchaSiteKey);
+  const hCaptchaRef = useRef<HCaptcha>(null);
+
+  // Hopefully there is a better way to do this :(
+  const formEventRef = useRef<FormEvent<HTMLFormElement>>(null);
+
+  if (captchaEnabled && !hCaptchaSiteKey) {
+    logMessage.error("hCaptcha: hCaptchaSiteKey is missing");
+    return null;
+  }
+
+  // Verify the hCAPTCHA token on the server side.
+  const verify = async (token: string) => {
+    try {
+      const success = await verifyHCaptchaToken(token);
+      if (success) {
+        logMessage.info(`hCaptcha: success`);
+        onSubmit(formEventRef.current as FormEvent<HTMLFormElement>);
+      } else if (blockableMode) {
+        logMessage.info(`hCaptcha: failed and submission blocked`);
+        router.push(`/${lang}/unable-to-process`);
+      } else {
+        logMessage.info(`hCaptcha: failed and submission allowed`);
+        onSubmit(formEventRef.current as FormEvent<HTMLFormElement>);
+      }
+    } catch (err) {
+      logMessage.error(`hCaptcha: system error ${err}`);
+      // Don't block the user from submittion on a system error (our fault not theirs)
+      onSubmit(formEventRef.current as FormEvent<HTMLFormElement>);
+    }
+  };
+
+  if (!captchaEnabled) {
+    return (
+      <form
+        {...(id ? { id } : {})}
+        {...(dataTestId ? { "data-test-id": dataTestId } : {})}
+        {...(className ? { className } : {})}
+        {...(noValidate ? { noValidate } : {})}
+        /**
+         * method attribute needs to stay here in case javascript does not load
+         * otherwise GET request will be sent which will result in leaking all the user data
+         * to the URL
+         */
+        method="POST"
+        onSubmit={(e) => {
+          e.preventDefault();
+          onSubmit(e);
+        }}
+      >
+        {children}
+      </form>
+    );
+  }
+
+  // see https://github.com/hCaptcha/react-hcaptcha
+  return (
+    <form
+      {...(id ? { id } : {})}
+      {...(dataTestId ? { "data-test-id": dataTestId } : {})}
+      {...(className ? { className } : {})}
+      {...(noValidate ? { noValidate } : {})}
+      method="POST"
+      onSubmit={(e) => {
+        e.preventDefault();
+        formEventRef.current = e;
```

Comment:
The forms-form currently doesn't use the forms event and I'm not sure any future form will. It would still be nice to get this sorted in advance though. I suspect a future form will.
I wonder if there is a better way to do this. Like some kind of dynamic event binding but that wouldn't be much better than a ref.